src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -1576,6 +1577,22 @@ impl ThinPool {
         }
         Ok(changed)
     }
+
+    pub fn unset_fs_origin(&mut self, fs_uuid: FilesystemUuid) -> StratisResult<bool> {
+        let changed = {
+            let (_, fs) = self.get_mut_filesystem_by_uuid(fs_uuid).ok_or_else(|| {
+                StratisError::Msg(format!("No filesystem with UUID {fs_uuid} found"))
+            })?;
+            fs.unset_origin()
+        };
+        let (name, fs) = self
+            .get_filesystem_by_uuid(fs_uuid)
+            .expect("Looked up above.");
```

Comment:
I don't see any reason for this not to be inside the `if changed ...` block. That would avoid looking it up if nothing has changed.